compile.c

```diff
@@ -6660,12 +6660,7 @@ compile_redo(rb_iseq_t *iseq, LINK_ANCHOR *const ret, const NODE *const node, in
 	    ADD_INSN(ret, line, putnil);
 	}
     }
-    else if (iseq->body->type == ISEQ_TYPE_EVAL) {
-      redo_in_eval:
-	COMPILE_ERROR(ERROR_ARGS "Can't escape from eval with redo");
-	return COMPILE_NG;
-    }
-    else if (ISEQ_COMPILE_DATA(iseq)->start_label) {
+    else if (iseq->body->type == ISEQ_TYPE_EVAL && ISEQ_COMPILE_DATA(iseq)->start_label) {
```

Comment:
Are these expressions equivalent?
Seems I'm missing something...

Reply:
Ah, this is a mistake.  Thank you.  It should be:

```suggestion
    else if (iseq->body->type != ISEQ_TYPE_EVAL && ISEQ_COMPILE_DATA(iseq)->start_label) {
```

Reply:
https://github.com/ruby/ruby/pull/3247/commits/d472f2652b2d39d359741d788fa4a23384f9aa32